### ISSUE TITLE
Issues with Google Play Music player not loading

### ISSUE THREAD

**Issue by shaneg8514** [MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-#3679]
Hello-

I wrote a short while ago about GPMP hanging on the load screen on Linux Mint 19.1.  I was directed toward the Circle CI via a direct link to a new build. That did the trick.

Recently, I just got a new laptop, which I  promptly installed Mint 19.2 on. Since doing so, even using the link pointed toward in #3660 , I am hanging on the loading music library screen.

I have poked around in the Circle CI, looking for an updated artifact, but, and this might just be me being dense, I cannot find the artifacts link alluded to in the readme.

Any insight? Thanks so much

Shane

**Checklist:**

* [x] I have checked that there are no issues with similar or the same content

  **YOU SHOULD CHECK CLOSED ISSUES ASWELL**
* [x] I have checked the FAQ (https://github.com/MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-/wiki/FAQ) and the answer I am looking for is not there
* [x] I have double checked and can reproduce the issue

**OS:**

**GPMDP Version:**

**Issue Descriptions:**

**Steps to Reproduce:**

**Comment by jostrander** [comment]
Unfortunately the artifacts on their website are behind a login screen. There are various links to the artifact builds in the issues however.

**Comment by jostrander** [comment]
I'm closing this as it's a general question related to #3660.

**Comment by shaneg8514** [comment]
Thanks for the reply.

The problem is the only link for Debian based systems in #3660 does not work in this instance. (It worked previously however on my older computer).

I did use my Github account to log into CI, still no luck finding the solution.

Can you suggest a way forward?

**Comment by jostrander** [comment]
https://app.circleci.com/jobs/github/MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-/3718/artifacts

These are the artifacts from the latest version

**Comment by shaneg8514** [comment]
Hello again,

Thanks for the links. Unfortunately, I'm still stuck on the loading screen. I tried the 64bit artifact from both the first and second links (although by file names, it seems they may be the same).

Is there anything I can provide you from my system setup that might be of use?

Shane

**Comment by jostrander** [comment]
`CTRL+SHIFT+G` -> type in (no quotes) `DEV_MODE` -> restart -> `CTRL+SHIFT+i` -> Screenshot of any errors in this window. Type in `openGPMDevTools()` `enter`. Screenshot any errors in this window.

**Comment by shaneg8514** [comment]
Hi Jesse-

After launching DEV_MODE & restarting, I was able to start the player and it worked as it should.

I didn't spot any errors per se in the ctrl/shift/i configuration (but grabbed a shot). Under the openGPMDevTools() I caught several errors (across two images):

![ctrlshifti](https://user-images.githubusercontent.com/57635171/69082295-6ee77c00-0a05-11ea-9f20-7c47a72752bd.png)
![error1](https://user-images.githubusercontent.com/57635171/69082296-6ee77c00-0a05-11ea-8df6-c54ddb3a2f94.png)
![error2](https://user-images.githubusercontent.com/57635171/69082298-6f801280-0a05-11ea-9967-3924f28c18a1.png)


Shane

**Comment by jostrander** [comment]
That looks like you might have some kind of UI update that the rest of us don't have, considering its complaining about UI being missing, and you are loading listen_v2.js which I don't have yet. Is there anything different in the browser for you?
I also see an error related to mDNS, that could be causing some problems as well.

**Comment by shaneg8514** [comment]
Hi Jesse-

Sorry for my slow reply. I sent my last message literally running out the door.

The browser in Dev mode looks pretty normal (there is an 'I'm feeling lucky' button which I don't recall from last time. Other than that, everything seems normal as far as I can tell.

The UI I'm working off of is an XFCE desktop on Mint 19.2, I don't think I have any Gnome or KDE packages installed (other than what might come from installing Stata's graphical library or gedit). I'm not quite sure what mDNS is. 

Is it 'safe' to run it in dev mode on the regular? The player is much nicer than using a browser tab. Of course, I'm happy to do anything I can to help you all moving forward.

**Comment by jostrander** [comment]
Sorry, by UI I was referring to the GPM UI, not the desktop UI. Either way, if it works in Dev mode then it should work outside of dev mode. mDNS is the DNS service used for zero configuration discovery for Chromecast, so it appears to be broken for you, or you have a device without a name on your network.

**Comment by shaneg8514** [comment]
Hi Jesse-

I started Googling mDNS and Chromecast. Mint 19.2 has the mDNS package by default. But, on a whim, I installed chromium-browser (thinking it would carry with it the things needed to make Chromecast work). Indeed, it does: After a reboot, GPDMP works as normal. Perhaps there is some dependency in Chromium that the player also relies upon?

Thanks again-- I really appreciate it and I hope you have a great night.

**Comment by Barteks2x** [comment]
I also have the same issue on gentoo after updates and updating to latest GPMDP from the above link doesn't solve the issue, but the errors appear to be a little different and it DOESN'T work in dev mode and I don't have a workaround, other than using it in a browser without the app.
Screenshot of the errors window:
![image](https://user-images.githubusercontent.com/1349112/69359241-8a15ee00-0c88-11ea-98d5-5d0c3a3cae25.png)

**Comment by jostrander** [comment]
Your issue is the same issue that was fixed in #3660. I'm not sure how you get the application in gentoo, but if you can get the latest source and run it, it is fixed.

**Comment by Barteks2x** [comment]
Gentoo generally is source based so it should in theory be easy for me to run from latest sources, but so far noone tried to make a source based GPMDP ebuild, so there is only a `-bin` ebuild in some unofficial overlay (I also have my own live ebuild with version from https://3718-40008106-gh.circle-artifacts.com/0/home/circleci/project/dist/installers/debian/google-play-music-desktop-player_4.6.1_amd64.deb, making an ebuild from deb package is usually not that hard).

I tried to get GPMDP built from source, but that seems to require electron, + a bunch of other dependencies that there are no ebuilds for so I would have to make them and at that time I just gave up and got the bin package. At this moment I don't have enough time to get it done (but I may try, maybe this weekend). Is there some prebuilt binary of the latest linux version?
